buildkite/scripts/changelog.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# This script checks if a changelog entry is required for a given pull request.
+# It compares the current commit with the base branch of the pull request and looks for changes
+# in the specified trigger files. If changes are found, it checks if the required changelog entry
+# is present. If not, it exits with an error. Can be bypassed by a !ci-bypass-changelog comment.
+
+# Usage:
+#   ./changelog.sh --trigger <trigger> --required-change <required-change>
+#
+# Options:
+#   --trigger: The trigger to look for in the diff
+#   --required-change: The required change to look for in the diff
+#   -h, --help: Display help message
+#
+# Example:
+#   ./changelog.sh --trigger 'src/daemon' --required-change 'changes/1234-new-feature.md'
+
+TRIGGER=""
+REQUIRED_CHANGE=""
+# List of users who can bypass the changelog check by commenting !ci-bypass-changelog
+# separated by space
+GITHUB_USERS_ELIGIBLE_FOR_BYPASS="amc-ie mrmr1993"
```

Comment:
@dkijania could you also add team leads?

Handles: `deepthiskumar Trivo25 45930 SanabriaRusso nicc georgeee`